### PR TITLE
Write lesson comparing compile-time errors and runtime errors

### PR DESCRIPTION
**Pull request by kcleland0818**

### CODE REVIEW

**Review by linda-seiter — APPROVED** [review]
Nice examples!

File: README.md [hunk 1]
```diff
@@ -2,31 +2,102 @@
...
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+![use-variable-without-initialization-intellij](https://curriculum-content.s3.amazonaws.com/java-mod-1/variables/intellij-use-variable-without-initialization.png)
 
-## Conclusion
+We see IntelliJ warning us of an error. When we try to run the program, Java
+will first try to compile the source code. But it stops before it can even
+proceed running it:
```
Comment (developerjaym) [inline README.md]: Consider adding 'to' between 'proceed' and 'running'

File: README.md [hunk 2]
```diff
@@ -2,31 +2,102 @@
...
+  ![missing-curly-bracket](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-missing-curly-bracket.png)
+
+- Missing terminator.
+
+  ![missing-semicolon](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-missing-terminator.png)
+
+- And other syntax type errors.
+
+  ![syntax-errors](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-syntax-errors.png)
+
```
Comment (developerjaym) [inline README.md:53]: Consider adding an explanation of why compile-time errors are really convenient for developers.
For example, in JavaScript or Python, a missing comma or misspelled variable name can lead to an hour of frustration.
I've noticed a lot of new developers don't understand all the ways their tools and languages are there to help them succeed with minimal pain.

File: README.md [hunk 3]
```diff
@@ -2,31 +2,102 @@
...
+compilation step and throws the exception at runtime:
+
+![arithmeticexception](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/runtime-error-divide-by-zero.png)
+
+Notice in this example that the program compiled and was able to start running
+successfully. It wasn't until the program tried to execute the
+`int notANumber = 9 / 0;` that the program failed to continue executing. This is
+an example of a runtime error.
+
+Notice in this example that the program compiled and was able to start running
```
Comment (developerjaym) [inline README.md]: Is an example image missing here?

**Review by developerjaym — CHANGES_REQUESTED** [review]
Looks like the last paragraph in runtime errors is referencing a missing example

**Review by linda-seiter — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kcleland0818) [reply README.md:53]: Good idea!

File: README.md (hunk 3 above)
Comment (kcleland0818) [reply README.md]: Oops! This was leftover when I refactored this. Thanks for catching. I removed it!

**Review by developerjaym — APPROVED** [review]

**Review by linda-seiter — APPROVED** [review]

**Pull request merged**